py/server/deephaven/column.py

```diff
@@ -32,47 +32,105 @@ def __repr__(self):
         return self.name
 
 
-@dataclass
-class Column:
-    """ A Column object represents a column definition in a Deephaven Table. """
-    name: str
-    data_type: DType
-    component_type: DType = None
-    column_type: ColumnType = ColumnType.NORMAL
+class Column(JObjectWrapper):
+    """A Column object represents a column definition in a Deephaven Table."""
+
+    j_object_type = _JColumnDefinition
+
+    def __init__(
```

Comment:
This includes that case:

```python
Column(j_column_definition=...)
```

And we use this construction as the basis for TableDefinition as a Mapping[str, Column].

It looks funky because we need to maintain compatibility with the existing constructor.

Reply:
We just agreed on a call to introduce a superclass called `ColumnDefinition`, maybe, in order to preserve the existing behavior and also provide the kind of wrapper we want users to use.